sentry-ruby/lib/sentry/transport.rb

```diff
@@ -136,14 +136,18 @@ def envelope_from_event(event)
       event_id = event_payload[:event_id] || event_payload["event_id"]
       item_type = event_payload[:type] || event_payload["type"]
 
-      envelope = Envelope.new(
-        {
-          event_id: event_id,
-          dsn: @dsn.to_s,
-          sdk: Sentry.sdk_meta,
-          sent_at: Sentry.utc_now.iso8601
-        }
-      )
+      envelope_headers = {
+        event_id: event_id,
+        dsn: @dsn.to_s,
+        sdk: Sentry.sdk_meta,
+        sent_at: Sentry.utc_now.iso8601
+      }
+
+      if event.is_a?(TransactionEvent) && event.dynamic_sampling_context
```

Comment:
Because we still support `async`, the event may be a hash. So we need to do these with the `event_payload` hash.

Reply:
hmm, but we should get rid of async soon too. I would rather not add more hash handling tbh, and it's fine if this only works for people on the new setup.

Reply:
I'm ok with that but I think we'll need to briefly mention this in the changelog.